src/widgets/RRule.tsx

```diff
@@ -0,0 +1,249 @@
+import * as React from 'react';
+import Container from './Container';
+import { TextField, Select, MenuItem, FormGroup, FormControlLabel, Checkbox } from '@material-ui/core';
+import RRule, { Options, Weekday } from 'rrule';
+
+import DateTimePicker from '../widgets/DateTimePicker';
+
+interface PropsType {
+  onChange: (rrule: string) => void;
+  rrule: string;
+}
+const frequency = {
+  yearly: 0,
+  monthly: 1,
+  weekly: 2,
+  daily: 3,
+};
+const weekdays = { 0: 'MO', 1: 'TU', 2: 'WE', 3: 'TH', 4: 'FR', 5: 'SA', 6: 'SU' };
+const months = {
+  1: 'Jan', 2: 'Feb', 3: 'Mar', 4: 'Apr', 5: 'May', 6: 'Jun', 7: 'Jul',
+  8: 'Aug', 9: 'Sep', 10: 'Oct', 11: 'Nov', 12: 'Dec',
+};
+const menueItemsMonths = Object.keys(months).map((key) => {
+  return (
+    <MenuItem key={key} value={key}>{months[key]}</MenuItem>
+  );
+});
+
+const bysetposWeekDay = Object.keys(weekdays).map((key) => {
+  return (
+    <MenuItem key={key} value={key}>{weekdays[key]}</MenuItem>
+  );
+});
+const menuItemsFrequency = Object.keys(frequency).map((key) => {
+  return (
+    <MenuItem key={key} value={frequency[key]}>{key}</MenuItem>
+  );
+});
+
+
+export default function RRuleEteSync(props: PropsType) {
+  const options = RRule.fromString(props.rrule).origOptions;
+
+  const updateRule = (newOptions: Partial<Options> = {}, weekdayToRemove: Weekday | null = null) => {
+    const updatedOptions = { ...options, ...newOptions };
+    if (Array.isArray(options.byweekday) && Array.isArray(newOptions.byweekday)) {
```

Comment:
Weekday to remove is a hack and shouldn't exist. And to an extent also this check with options.byweekday if it's an array.

Instead of weekday to remove, you should just pass the new array from the outside of this function. Intsead of isArray, just pass the new array to be used, like you would any other value.